### PR TITLE
feat(api): add internal API to create user

### PR DESCRIPTION
**Pull request by guangrui-cloudnatix**

### CODE REVIEW

**Comment by kkaneda** [comment]
Let me update the PR title to follow https://www.conventionalcommits.org/en/v1.0.0/#specification

File: server/internal/server/user.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	"github.com/llmariner/common/pkg/id"
+	v1 "github.com/llmariner/user-manager/api/v1"
+	"github.com/llmariner/user-manager/pkg/userid"
+	"github.com/llmariner/user-manager/server/internal/store"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+// CreateInternalUser creates a new user and related organization and project, and generates an API key.
```
Comment (kkaneda) [inline server/internal/server/user.go:16]: Could you add a TDDO comment on running all DB operations in a single transaction?

**Review by kkaneda — APPROVED** [review]
LG!

File: api/v1/user_manager_service.proto [hunk 2]
```diff
@@ -327,4 +333,7 @@ service UsersInternalService {
   }
   rpc ListProjectUsers(ListProjectUsersRequest) returns (ListProjectUsersResponse) {
   }
+
+  rpc CreateInternalUser(CreateInternalUserRequest) returns (google.protobuf.Empty) {
```
[the remark this answers is not in the document]
Comment (guangrui-cloudnatix) [reply api/v1/user_manager_service.proto:337]: In some case, this function creates a new user only, no org or project, e.g. back-filling users from an existing tenant. "CreateTenant" might be confusing?

File: server/internal/server/user.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+	orgs, err := s.store.ListAllOrganizations()
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "list organizations: %s", err)
+	}
+
+	var foundTenant bool
+	for _, org := range orgs {
+		if org.TenantID == req.TenantId {
+			foundTenant = true
```
[the remark this answers is not in the document]
Comment (kkaneda) [reply server/internal/server/user.go:46]: So this is intentional. Could you add a comment somewhere? Probably in the RPC definition.

[the remark this answers is not in the document]
Comment (guangrui-cloudnatix) [reply server/internal/server/user.go:46]: Good point, will return non-error for both cases.

**Pull request merged**